### DISCUSSION TITLE
Is it possible to run sd_dreambooth_extension using CPU only?

### DISCUSSION THREAD

**Discussion by peterhbg** [d8ahazard/sd_dreambooth_extension#156] [General]
Question above, and if no, what are the challenges of implementing a CPU only training method a.k.a. what are the limiting factors?  Because as far as I know, CPU only generation of images and merging ckpts are already implemented and the CPU only training check kind of suggests one can train the model using CPU only. 
As a side note, I really am up for waiting 40+ hours of training on CPU if I can get good results with this extension. I've tried dreambooth on Colab but the settings are really limiting compared to this extension, especially the ability to train the model using different settings at different times while maintaining a lifetime step count. That is simply the most amazing thing in the extension for me!
Thanks for the awesome tool, I hope I can manage to use it, be it on CPU or in a new laptop that can handle the load on the GPU.

**Comment by FlexTestHD** [comment]
First of all: yes, it's possible! :) 
I've tested the CPU only option yesterday with a set of 14 images, a training step count of 1100, a learning rate of 0.000001 and also 200 classification images - it roughly took me around 19 hours (on a Ryzen 5800x). I just disabled  "Use 8bit Adam" and "Mixed precision" in the advaned settings tab, after that, it worked for me.

There is also a bug at the current time, so the training is working properly, but it doesn't generate a .ckpt-file. A user in another thread mentioned this https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/4639#issuecomment-1312670524, so I deleted the 130-line and it worked again. But that's probably fixed any time soon. I'm also very thankful for the tool and the possibility to only use CPU (because my 1060 with 3GB VRAM is not quite as committed to the cause). It does take time and my results aren't that great right now (because tinkering isn't that easy with such times) - but it's working and this is, what counts. :D

**Comment by xhost007** [comment]
Yes, finally got it to work with awesome results training on a person.
I use 1200 steps LR=2E-6, 1500 normalization pictures (from Aitrepreneur repo), only 10 pictures in the training set. I used the auto-adjust button but rechecked "train text encoder" because it actually did not lead to very good results without it.
On my i7-13700K not OC, it took "only" 4 hours to train. 
ckpt file generated no problem here.
Next step my cat :-)

**Reply by FlexTestHD** [reply]
So glad to hear it worked so well! Also thanks for sharing your settings, will give it a try with that too. My cats will be looking forward to the model for sure. :D
